<?xml version="1.0"?>
<robot xmlns:xacro="http://ros.org/wiki/xacro" name="xarm">

    <!-- <xacro:property name="eef" value="eef_3_finger"/> -->
    <xacro:property name="eef" value="eef_2_finger"/>
    <xacro:arg name="sim_mode" default="true"/>
    <xacro:arg name="serial_port_esp32" default="/dev/ttyUSB0"/>

    <!-- <xacro:include filename="$(find xarm_description)/urdf/xarm.urdf"/> -->
    <xacro:include filename="$(find xarm_description)/urdf/xarm_v3.urdf"/>
    <link name="world"/>
    <xacro:arg name="initial_z" default="0.0"/>

    <joint name="world_joint" type="fixed">
        <parent link="world"/>
        <child link="xarm_base_link"/>
        <origin xyz="0.0 0.0 $(arg initial_z)" rpy="0.0 0.0 0.0"/>
    </joint>

    <xacro:include filename="$(find xarm_description)/urdf/ros2_control.arm.xacro" />
    <xacro:include filename="$(find xarm_description)/urdf/ros2_control_params.xacro" />
</robot>